src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -83,6 +87,8 @@ std::string GcsPlacementGroup::DebugString() const {
 }
 
 rpc::Bundle *GcsPlacementGroup::GetMutableBundle(int bundle_index) {
+  // Invalidate the cache.
+  cached_bundle_specs_.clear();
```

Comment:
hm... just a small question: Is it a bit brittle if we add other modified interfaces later?

Reply:
It could be. I think I can remove this caching logic after handing the second part of https://github.com/ray-project/ray/issues/18651 (which will fundamentally fix the issue). This PR is more for short term solution for the next release (I will make a PR for the fundamental fix next week)